blues/simulation.py

```diff
@@ -197,12 +197,19 @@ def chooseMove(self):
             correction_factor = (nc_state0['potential_energy'] - md_state0['potential_energy'] + alch_state1['potential_energy'] - nc_state1['potential_energy']) * (-1.0/self.nc_integrator.kT)
             log_ncmc = log_ncmc + correction_factor
 
-        if log_ncmc > randnum:
+        #check if potential energy is greater then precision and reject if that's the case
+        if abs(nc_state1['potential_energy']._value) > 2147483640.:
```

Comment:
@sgill2 - I'm not actually convinced this is the way to do it, as it's not that single precision can't store a floating point value larger than this number per se; it's that it can't measure differences that differ by more than the limits of single precision (at least as I understand it). In other words, I think large potential energies should be fine as long as they STAY large, since energy is always arbitrary up to some constant. It's if the difference in energy becomes larger than the precision (I think) that you have problems.

Can you justify why this particular choice? Did you try looking at energy DIFFERENCES instead?

My thought is that someone could do something funky to their system which made the total potential energy high (maybe, simulating a system consisting of purely repulsive spheres, or taking a charged system and applying an external potential, or....?) but where the system was otherwise "fine" (no large forces, etc.) and this could make it impossible to simulate such a system even though there are no precision issues.